crates/sui-core/src/authority.rs

```diff
@@ -1068,8 +1068,9 @@ impl AuthorityState {
         let storage_gas_price = protocol_config.storage_gas_price();
 
         let gas_object_id = ObjectID::random();
+        // give the gas object 2x the max gas to have coin balance to play with during execution
```

Comment:
Why 2x vs 1x?

Reply:
Because we give the transaction max gas, so the Coin would be empty during dev inspect, and it wouldn't really be usable for `SplitCoin` or anything